### PR TITLE
Support setting profiles-dir

### PR DESCRIPTION
**Pull request by ConstantinoSchillebeeckx**
### Description

Allows user to set `--profiles-dir` instead of needing to set the ENV `DBT_PROFILES_DIR` - setting this option is more flexible than the ENV because it can be done at the command line or as an INI option.

### Checklist

- [x] I read the [CONTRIBUTING](https://github.com/godatadriven/pytest-dbt-core/blob/main/CONTRIBUTING.md) guide and understand what's expected of me
- [x] I ran this code in development and it appears to resolve the stated issue
- [ ] I added tests, or tests are not required/relevant for this PR
- [x] I added an entry to the [CHANGELOG](https://github.com/godatadriven/pytest-dbt-core/blob/main/CHANGELOG.md) detailing the change of this PR

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 - Add example with column rename macro ([PR](https://github.com/godatadriven/pytest-dbt-core/pull/19))
 
+## [0.1.1] - 2023-02-22
```
Comment (JCZuurmond) [inline CHANGELOG.md:3]: You can remove this and add your entry on line two. The release script automatically adds the version number and date section header

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+
+- Expore `--profiles-dir` as option for setting profiles directory instead of using `DBT_PROFILES_DIR` environment variable
```
Comment (JCZuurmond) [inline CHANGELOG.md:5]:
```suggestion
- Add `--profiles-dir` as an option for setting the profiles directory
```

File: src/pytest_dbt_core/fixtures.py [hunk 3]
```diff
@@ -58,6 +59,12 @@ def config(request: SubRequest) -> RuntimeConfig:
     RuntimeConfig
         The runtime config.
     """
+
+    # only override profiles_dir if its set, otherwise keep DBT behavior
```
Comment (JCZuurmond) [inline src/pytest_dbt_core/fixtures.py:63]: I prefer to keep "why" comments only. This comment describes what I can read from the code

```suggestion
```

File: src/pytest_dbt_core/fixtures.py [hunk 4]
```diff
@@ -58,6 +59,12 @@ def config(request: SubRequest) -> RuntimeConfig:
...
+    profiles_dir = request.config.getoption("--profiles-dir")
+    if profiles_dir:
+        flags.PROFILES_DIR = profiles_dir
```
Comment (JCZuurmond) [inline src/pytest_dbt_core/fixtures.py:66]: Nicely done! I trust you found that this is the dbt way of setting the profiles directory?

**Review by JCZuurmond — CHANGES_REQUESTED** [review]
@ConstantinoSchillebeeckx: Thank you for your PR, it looks very well done! I added some minor changes. Though, the CI is causing some problems:

1. Could you rebase your branch on the main branch after fetching the changes? The isort pre-commit plugin caused problems, which are solved in the [latest release](https://github.com/PyCQA/isort/releases/tag/5.11.5). 
2. [It looks like](https://github.com/godatadriven/pytest-dbt-core/actions/runs/4276188135/jobs/7444092982) dbt requires a `profile` attribute to be set on the `Args`. This looks like a change done by dbt unrelated to your PR. However, since you are working on adding the profiles directory as a flag, maybe you could solve this?

**Comment by JCZuurmond** [comment]
@ConstantinoSchillebeeckx : I updated the testing matrix. [It shows](https://github.com/godatadriven/pytest-dbt-core/actions/runs/4276648450/jobs/7444887672) that the set-up fails since dbt 1.4.

File: src/pytest_dbt_core/fixtures.py (hunk 4 above)
Comment (ConstantinoSchillebeeckx) [reply src/pytest_dbt_core/fixtures.py:66]: Yep, you can see them doing it [here](https://github.com/dbt-labs/dbt-core/blob/c82795a99f13bcbe0a86e6084432702bbef214eb/core/dbt/main.py#L1225)

File: src/pytest_dbt_core/fixtures.py (hunk 3 above)
Comment (ConstantinoSchillebeeckx) [reply src/pytest_dbt_core/fixtures.py]: removed https://github.com/godatadriven/pytest-dbt-core/pull/25/commits/9c04c9242202d3573ee12a37a566f7a3bc60fcea

File: CHANGELOG.md (hunk 1 above)
Comment (ConstantinoSchillebeeckx) [reply CHANGELOG.md]: changed https://github.com/godatadriven/pytest-dbt-core/pull/25/commits/9c04c9242202d3573ee12a37a566f7a3bc60fcea

**Comment by ConstantinoSchillebeeckx** [comment]
@JCZuurmond could you kick CI off again please?

File: src/pytest_dbt_core/fixtures.py (hunk 4 above)
Comment (JCZuurmond) [reply src/pytest_dbt_core/fixtures.py:66]: Check, could you add an `abspath` like they do [there](https://github.com/dbt-labs/dbt-core/blob/c82795a99f13bcbe0a86e6084432702bbef214eb/core/dbt/main.py#L1222)?

And I am a bit confused about [dbt removing their main](https://github.com/dbt-labs/dbt-core/commit/c53c3cf181bf4facfb34ea5068d033bfc314eb72), where this piece of code is coming from. But if this works for now, let's merge this and tackle that later (if it is required).

File: setup.cfg [hunk 5]
```diff
@@ -79,7 +80,6 @@ setenv =
     PIP_DISABLE_PIP_VERSION_CHECK = 1
     COVERAGE_FILE = {env:COVERAGE_FILE:{toxworkdir}/.coverage.{envname}}
     {py27,pypy}: PYTHONWARNINGS=ignore:DEPRECATION::pip._internal.cli.base_command
-    DBT_PROFILES_DIR = {env:DBT_PROFILES_DIR:{toxinidir}/tests/dbt_project}
```
Comment (JCZuurmond) [inline setup.cfg:82]: Could you keep one of the tox env with `DBT_PROFILES_DIR` to test for both scenario's?

File: setup.cfg [hunk 6]
```diff
@@ -97,7 +97,6 @@ description = run the tests in docs
 setenv =
     PIP_DISABLE_PIP_VERSION_CHECK = 1
     {py27,pypy}: PYTHONWARNINGS=ignore:DEPRECATION::pip._internal.cli.base_command
-    DBT_PROFILES_DIR = {env:DBT_PROFILES_DIR:{toxinidir}/docs/source/_static/dbt_project}
```
Comment (JCZuurmond) [inline setup.cfg:100]: You have to add the `--profiles-dir` flag in the `pytest` call a couple lines below

**Review by JCZuurmond — CHANGES_REQUESTED** [review]
I have added some commits to the tox call

File: setup.cfg (hunk 6 above)
Comment (ConstantinoSchillebeeckx) [reply setup.cfg:100]: good catch https://github.com/godatadriven/pytest-dbt-core/pull/25/commits/a5420f3f1feac5d753d2cae1cd002bfa51e8862f

File: setup.cfg (hunk 5 above)
Comment (ConstantinoSchillebeeckx) [reply setup.cfg]: done https://github.com/godatadriven/pytest-dbt-core/pull/25/commits/a5420f3f1feac5d753d2cae1cd002bfa51e8862f

File: src/pytest_dbt_core/fixtures.py (hunk 4 above)
Comment (ConstantinoSchillebeeckx) [reply src/pytest_dbt_core/fixtures.py]: sure thing: https://github.com/godatadriven/pytest-dbt-core/pull/25/commits/1ddb9791e1ddfb1a157dd31d70d7d02c9610ec68

it looks like `1.5` re-arranged a bunch of code (like you mention, e.g. removing main) - I don't know the full extent of those changes, but I'm suspecting it'll involve a bit of work within this library to support it. they've got a release-candidate out so we could start poking around

Comment (JCZuurmond) [reply src/pytest_dbt_core/fixtures.py]: You're very much welcome to have a look around! And let's merge this PR first.

**Review by JCZuurmond — APPROVED** [review]
@ConstantinoSchillebeeckx : LGTM! Thanks for your contribution. Will you merge the PR? I guess you can after resolving the last conversation.

**Comment by ConstantinoSchillebeeckx** [comment]
@JCZuurmond branch protection or repo setup doesn't allow me to merge this - feel free to do it.

**Pull request merged**